kernel/src/sched/stats/loadavg.rs

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! This module implements the CPU load average calculation.
+//!
+//! Reference: <https://github.com/torvalds/linux/blob/master/kernel/sched/loadavg.c>
+
+use ostd::{arch::timer::TIMER_FREQ, sync::RwLock, timer};
+
+/// nr of bits of precision
+pub const FSHIFT: u64 = 11;
+/// 1.0 as fixed-point
+pub const FIXED_1: u64 = 1 << FSHIFT;
+/// 5 sec intervals
+pub const LOAD_FREQ: u64 = 5 * TIMER_FREQ + 1;
+/// 1/exp(5sec/1min) as fixed-point
+pub const EXP_1: u64 = 1884;
+/// 1/exp(5sec/5min)
+pub const EXP_5: u64 = 2014;
+/// 1/exp(5sec/15min)
+pub const EXP_15: u64 = 2037;
+
+/// Load average of all CPU cores.
+///
+/// The load average is calculated as an exponential moving average of the load
+/// over the last 1, 5, and 15 minutes.
+#[derive(Clone)]
+pub struct LoadAverage {
+    load: [u64; 3],
+    last_update: u64, // jiffies
+}
+
+impl LoadAverage {
+    /// Creates a new `LoadAverage`.
+    pub const fn new() -> Self {
+        Self {
+            load: [0, 0, 0],
+            last_update: 0,
+        }
+    }
+}
+
+/// The global load average.
+pub static LOAD_AVG: RwLock<LoadAverage> = RwLock::new(LoadAverage::new());
+
+/// Returns the calculated load average of the system.
+pub fn get_loadavg() -> [u64; 3] {
+    let mut load = LOAD_AVG.read().clone().load;
+
+    load[0] += FIXED_1 / 200;
+    load[1] += FIXED_1 / 200;
+    load[2] += FIXED_1 / 200;
+
+    load
+}
+
+/// Returns the calculated load average of the system as floating point numbers.
+pub fn get_loadavg_float() -> [f64; 3] {
```

Comment:
In kernel programming, we cannot use floating point numbers. For one thing, when doing user-kernel switching (e.g., system calls), the floating pointer portion of the CPU state is not saved for efficiency. So the kernel should NOT touch the floating pointer state by doing floating pointer arithmetic.

Furthermore, after importing the `fixed` crate, the fixed point types provided by it support `Debug` and `Display` traits (I guess). So you do not have to convert fixed-point numbers to floating-point ones in order to display them as a string.